### DISCUSSION TITLE
Scanning Medications

### DISCUSSION THREAD

**Discussion by seddinger** [marsupialproject/marsupial-emr#4] [General]
Hi, I can't get scanning to work when we scan a medication. We have the order set and it's on the MAR to be scanned. When we go to scan the medication nothing happens. The medication is in the med library with the correct barcode. Just curious, how does Marsupial know where to look in the med library, when the medlibrary DB isn't connected during the scanning process. Not for sure what I'm doing wrong.

When adding your own medications to the medication DB where are you getting the barcode for the name of the drug.  Is there a website that you get from or are you getting the barcode from the medication package?

After editing with editor, I click on "Refresh" and Marsupial EMR doesn't refresh with the newest information, and why?

**Comment by marsupialproject** [comment]
> Hi, I can't get scanning to work when we scan a medication. We have the order set and it's on the MAR to be scanned. When we go to scan the medication nothing happens. The medication is in the med library with the correct barcode. Just curious, how does Marsupial know where to look in the med library,

This is explained in the Editor video and wiki. The generic name of the drug is matched between what's physically scanned, the MAR entry name, and the EMR order description. All three names must match exactly, including case. The MAR entry's "Order ID" must match the EMR order's Order ID. The route must also match (i.e. Enteral or Parenteral).

If meds aren't scanning, then the cause is likely a problem with one or more of these. You can refer to the example database for what a working database and medlibrary looks like.

> when the medlibrary DB isn't connected during the scanning process.

The medlibrary is connected automatically. It will load the medlibrary.db in the Marsupial EMR directory. You should really only have one medlibrary, so you are not able to choose which to load.

> When adding your own medications to the medication DB where are you getting the barcode for the name of the drug. Is there a website that you get from or are you getting the barcode from the medication package?

As said in the Editor video, I use manufactured "educational" meds from vendors like Demo Dose and Mock Meds. The meds come barcoded, so I use those barcodes. It makes it very easy and they are reasonably priced. For meds they don't have, I Google Image search for a drug label, print it, and use that barcode. Just be sure to label them as "not for human use" or something like that for safety reasons. You could use a drug label database like DailyMed as well (https://dailymed.nlm.nih.gov/dailymed/index.cfm).

> After editing with editor, I click on "Refresh" and Marsupial EMR doesn't refresh with the newest information, and why?

Because you are editing the original database with the Editor, while the EMR is using the shadow copy it creates from the original. If you want to refresh edits you are concurrently making, you will need to close and reopen Marsupial EMR, or manually reload the database in the File menu.

**Reply by marsupialproject** [reply]
To further clarify on the scanning match process:

1. Patient wristband is scanned. It will compare the scanned barcode, which is the MRN, to the open patient's MRN. If this matches, it will say "Patient Match" and you can scan meds.
2. A medication is scanned. It will look for that barcode in the medlibrary. If found, it will read the med name and route and popup an order selection box.
3. The order selection box will populate all MAR entries matching the scanned med's name and route. For example, if you scanned a vial of furosemide, and the order is for oral furosemide, that will not match. If there are two IV push furosmide MAR entries (e.g. 0900 and 1800), they will both show up in the list.

**Reply by seddinger** [reply]
The med library isn't connecting automatically and the popup box for "Scanned Medication" comes up, but no "Select Order to Administer" comes up.  The med name is spelled correctly, and lowercase, barcode matches, and route matches "Order" number.

**Reply by marsupialproject** [reply]
Attach or email me your entire "Marsupial EMR" folder as a zip file, along with the patient database and I'll take a look at it

**Comment by seddinger** [comment]
Office 365 and Gmail is blocking from sending the file.

Samuel Eddinger
IT Supp Assoc - Simul Lab Tech
Ninth Street Hall 2020
217-581-7031
***@***.******@***.***>

[45176785] The Gail & Richard Lumpkin
        School of Nursing

**Reply by marsupialproject** [reply]
Try sending just the two database files then

**Comment by seddinger** [comment]
I got the medications to come up, my fault misspellings of the medications in the "Order".  The only issue now, is the med library not automatically loading and I go to "Select Order to Administer", I pick "metoprolol tartrate 50 mg due at "12:00" and doesn't register the time on the "MAR @ 12:00".  I did the same for "furosemide", still didn't register my scanning time in green on the MAR.

Thanks,
Samuel Eddinger


***@***.***
Samuel Eddinger
IT Supp Assoc - Simul Lab Tech
Ninth Street Hall 2020
217-581-7031
***@***.******@***.***>

[45176785] The Gail & Richard Lumpkin
        School of Nursing

**Reply by marsupialproject** [reply]
I'm not sure what to tell you on that one. That shouldn't be possible, unless you've somehow blocked Marsupial EMR from editing the shadow database it's created. The program would probably crash if that was the case, so you likely have something configured incorrectly. I'll offer again to take a look at your files.

**Comment by seddinger** [comment]
I got everything working, except for the med library not automatically opening.

I really appreciate all your help!!

Thanks,
Samuel Eddinger

Samuel Eddinger
IT Supp Assoc - Simul Lab Tech
Ninth Street Hall 2020
217-581-7031
***@***.******@***.***>

[45176785] The Gail & Richard Lumpkin
        School of Nursing

**Reply by marsupialproject** [reply]
No problem, but I am unable to help you without your database files.

**Comment by eddingers** [comment]
Hi, I'm trying to use Instructor, but keep getting this error.  I attached the error that I'm receiving.

![Marsupial Instructor Error](https://github.com/user-attachments/assets/ba1a02a3-5f38-4981-827a-eb6799a5a8c7)

**Reply by marsupialproject** [reply]
Make a new thread if you're having a new issue

**Comment by eddingers** [comment]
Hi,

Could you be more specific on what you mean by "Make a new thread if you're having a new issue"?

Thanks,
Samuel Eddinger

**Reply by marsupialproject** [reply]
https://docs.github.com/en/discussions/quickstart#creating-a-new-discussion
